### PR TITLE
fix(api): support deferred input to `ibis.desc` and `ibis.asc`

### LINKED ISSUE #4705: TypeError when calling `.order_by(ibis.desc(ibis._.x))`
Context: the `ibis._` deferred attribute resolver was introduced in #3804. However it does not always interact with other ibis constructs as one would have expected:

Reproducer:

The following works as expected (ascending `order_by` on deferred column name):

```python
>>> import pandas as pd
... import ibis
... from ibis import _
... 
... table = ibis.memtable(pd.DataFrame({"x": list("abbac"), "y": range(5)}))
... (
...    table
...    .group_by(_.x)
...    .aggregate(mean_y=_.y.mean())
...    .order_by(_.mean_y)
... ).execute()
   x  mean_y
0  a     1.5
1  b     1.5
2  c     4.0
```

but combining this with `ibis.desc` leads to:

```python
>>> (
...    table
...    .group_by(_.x)
...    .aggregate(mean_y=_.y.mean())
...    .order_by(ibis.desc(_.mean_y))
... ).execute()
Traceback (most recent call last):
  Input In [38] in <cell line: 2>
    .order_by(ibis.desc(_.mean_y))
  File ~/code/ibis/ibis/expr/types/relations.py:479 in order_by
    return self.op().order_by(by).to_expr()
  File ~/code/ibis/ibis/expr/operations/relations.py:474 in order_by
    keys = rlz.tuple_of(rlz.sort_key_from(rlz.just(self)), sort_exprs)
  File ~/mambaforge/envs/dev/lib/python3.10/site-packages/toolz/functoolz.py:304 in __call__
    return self._partial(*args, **kwargs)
  File ~/code/ibis/ibis/common/validators.py:166 in container_of
    return type(inner(item, **kwargs) for item in arg)
  File ~/code/ibis/ibis/common/validators.py:166 in <genexpr>
    return type(inner(item, **kwargs) for item in arg)
  File ~/code/ibis/ibis/expr/rules.py:88 in __call__
    result = super().__call__(*args, **kwargs)
  File ~/mambaforge/envs/dev/lib/python3.10/site-packages/toolz/functoolz.py:304 in __call__
    return self._partial(*args, **kwargs)
  File ~/code/ibis/ibis/expr/rules.py:126 in sort_key_from
    key = function_of(table_ref, key, **kwargs)
  File ~/code/ibis/ibis/expr/rules.py:88 in __call__
    result = super().__call__(*args, **kwargs)
  File ~/mambaforge/envs/dev/lib/python3.10/site-packages/toolz/functoolz.py:304 in __call__
    return self._partial(*args, **kwargs)
  File ~/code/ibis/ibis/expr/rules.py:518 in function_of
    arg = fn.resolve(arg)
  File ~/code/ibis/ibis/expr/deferred.py:32 in resolve
    resolved_expr = self.resolve(expr)
  File ~/code/ibis/ibis/expr/deferred.py:32 in resolve
    resolved_expr = self.resolve(expr)
  File ~/code/ibis/ibis/expr/deferred.py:37 in resolve
    return func(resolved_expr, *resolved_args, **resolved_kwargs)
  File ~/code/ibis/ibis/expr/deferred.py:49 in <lambda>
    lambda expr, *args, **kwargs: expr(*args, **kwargs),
TypeError: 'FloatingColumn' object is not callable
```

Replacing the deferred resolver by the str name works:

```python
>>> (
...    table
...    .group_by(_.x)
...    .aggregate(mean_y=_.y.mean())
...    .order_by(ibis.desc("mean_y"))
... ).execute()
   x  mean_y
0  c     4.0
1  a     1.5
2  b     1.5
```

### PR DESCRIPTION
**Pull request by cpcloud**
Fixes #4705.

### CODE REVIEW

**Review by jcrist — APPROVED** [review]
:+1:

File: ibis/tests/expr/test_value_exprs.py [hunk 1]
```diff
@@ -1649,3 +1649,12 @@ def test_case_rlz_incompatible_error(table):
         table.g.case().when(case == result, result).end()
 
     assert "a:int8 and Literal(foo):string" in str(exc_info.value)
+
+
+@pytest.mark.parametrize("func", [ibis.asc, ibis.desc])
+def test_group_by_order_by_desc_deferred(func):
```
Comment (saulpw) [inline ibis/tests/expr/test_value_exprs.py:1655]: remove `_desc` in test func name

Comment (cpcloud) [reply ibis/tests/expr/test_value_exprs.py]: Done!

**Comment by ogrisel** [comment]
Thanks!

**Pull request merged**